src/views/Auth/Login.jsx

```diff
@@ -13,42 +13,54 @@ export default function Login() {
 
   // The `from` property of `location.state` gives us
   // the URL to redirect to after logging in.
+  //This line takes the user to their original desired page after login
   const { from } = location.state || { from: { pathname: '/' } };
 
   const handleLogin = (event) => {
     event.preventDefault();
     const loginWasSuccessful = auth.login(formState.email, formState.password);
 
-    // TODO: If login was unsuccessful, set an error with a message
-    // to display to the user that their login failed.
-    //
-    // If login was successful, use the history hook
-    // from React Router to replace the current URL with the URL
-    // we need to redirect to.
-    // See https://v5.reactrouter.com/web/api/history for the appropriate method to use
+    if(loginWasSuccessful) {
+      history.replace(from);
```

Comment:
While your tests are working so this line technically works, pay close attention here. The default value of "from" is an object with a property that contains the path. To access that, you'd need to use object notation. In this simple slimmed-down case it's not an issue but something you should be keenly aware of so as to avoid conflicts in the future.